src/index.js

```diff
@@ -1,32 +1,44 @@
-var XMLHttpRequest = require("xmlhttprequest").XMLHttpRequest;
+/* var XMLHttpRequest = require("xmlhttprequest").XMLHttpRequest; // Linea comentada para correr en navegador desde index.html */
 
 var API = 'https://rickandmortyapi.com/api/character/';
-var xhttp = new XMLHttpRequest();
 
-function fetchData(url_api, callback) {
-  xhttp.onreadystatechange = function (event) {
-    if (xhttp.readyState === '4') {
-      if (xhttp.status == 200)
-        callback(null, xhttp.responseText);
-      else return callback(url_api);
+const fetchData = url_api => {
+  let xhttp = new XMLHttpRequest()
+  return new Promise((resolve, reject) => {
+    xhttp.onreadystatechange = event => {
+      if (xhttp.readyState === 4) {
+        if (xhttp.status == 200) {
+          resolve(JSON.parse(xhttp.responseText))
+        } else {
+          reject(url_api)
```

Comment:
Te recomiendo ser mas claros en el reject, esto te permite identificar el error mas fácil. 

```suggestion
          reject(new Error("Algo salió mal."));
```